staging/src/k8s.io/code-generator/kube_codegen.sh

```diff
@@ -665,3 +665,95 @@ function kube::codegen::gen_client() {
             "${input_pkgs[@]}"
     fi
 }
+
+# Generate register code
+#
+# USAGE: kube::codegen::gen_registers [FLAGS] <input-dir>
+#
+# <input-dir>
+#   The root directory under which to search for Go files which request code to
+#   be generated.  This must be a local path, not a Go package.
+#
+#   See note at the top about package structure below that.
+#
+# FLAGS:
+#
+#   --boilerplate <string = path_to_kube_codegen_boilerplate>
+#     An optional override for the header file to insert into generated files.
+#
+function kube::codegen::gen_registers() {
+    local in_dir=""
+    local boilerplate="${KUBE_CODEGEN_ROOT}/hack/boilerplate.go.txt"
+    local v="${KUBE_VERBOSE:-0}"
+
+    while [ "$#" -gt 0 ]; do
+        case "$1" in
+            "--boilerplate")
+                boilerplate="$2"
+                shift 2
+                ;;
+            *)
+                if [[ "$1" =~ ^-- ]]; then
+                    echo "unknown argument: $1" >&2
+                    return 1
+                fi
+                if [ -n "$in_dir" ]; then
+                    echo "too many arguments: $1 (already have $in_dir)" >&2
+                    return 1
+                fi
+                in_dir="$1"
+                shift
+                ;;
+        esac
+    done
+
+    if [ -z "${in_dir}" ]; then
+        echo "input-dir argument is required" >&2
+        return 1
+    fi
+
+    (
+        # To support running this from anywhere, first cd into this directory,
+        # and then install with forced module mode on and fully qualified name.
+        cd "${KUBE_CODEGEN_ROOT}"
+        BINS=(
+            register-gen
+        )
+        # shellcheck disable=2046 # printf word-splitting is intentional
+        GO111MODULE=on go install $(printf "k8s.io/code-generator/cmd/%s " "${BINS[@]}")
+    )
+    # Go installs in $GOBIN if defined, and $GOPATH/bin otherwise
+    gobin="${GOBIN:-$(go env GOPATH)/bin}"
+
+    # Register
+    #
+    local input_pkgs=()
+    while read -r dir; do
+        pkg="$(cd "${dir}" && GO111MODULE=on go list -find .)"
+        input_pkgs+=("${pkg}")
+    done < <(
+        ( kube::codegen::internal::grep -l --null \
+            -e '+groupName=' \
```

Comment:
All of th eother greps have been changed to something like `-e '^\s*//\s*+genclient'` - I assume something similar applies here?

Reply:
As the Golang source code of register-gen indicates it extracts information from files with `+groupName` comment tag:
https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/code-generator/cmd/register-gen/generators/targets.go#L84-L90

Reply:
I meant that the anchoring could be more specific.  We found some places that were matching on other comments which included `+genclient` but were not actually the declaration for "please generate me a client".  E.g.

We want to match :

```
        // +genclient
```

but not:

```
        // Blah blah blah, and then we include +genclient in the code and blah blah blah
```

Reply:
Understood

Reply:
Fixed as suggested, please take a look if you have any time, thanks. :-)